x-pack/plugins/security_solution/public/common/components/event_details/insights/related_alerts_by_process_ancestry.tsx

```diff
@@ -167,18 +168,57 @@ const ActualRelatedAlertsByProcessAncestry: React.FC<{
   eventId: string;
   timelineId?: string;
 }> = ({ alertIds, eventId, timelineId }) => {
+  const dataProviderLimit = 5;
   const dataProviders = useMemo(() => {
     if (alertIds && alertIds.length) {
-      return alertIds.reduce<DataProvider[]>((result, alertId, index) => {
-        const id = `${timelineId}-${eventId}-event.id-${index}-${alertId}`;
-        result.push(getDataProvider('_id', id, alertId));
-        return result;
-      }, []);
+      if (alertIds.length > dataProviderLimit) {
+        return null;
+      } else {
+        return alertIds.reduce<DataProvider[]>((result, alertId, index) => {
+          const id = `${timelineId}-${eventId}-event.id-${index}-${alertId}`;
+          result.push(getDataProvider('_id', id, alertId));
+          return result;
+        }, []);
+      }
     }
     return null;
-  }, [alertIds, eventId, timelineId]);
+  }, [alertIds, eventId, timelineId, dataProviderLimit]);
+
+  const filters: Filter[] | null = useMemo(() => {
+    if (alertIds && alertIds.length && alertIds.length >= dataProviderLimit) {
+      return [
+        {
+          meta: {
+            alias: 'Process Ancestry Alert IDs',
+            type: 'phrases',
+            key: '_id',
+            params: [...alertIds],
+            negate: false,
+            disabled: false,
+            value: alertIds.join(),
+          },
+          query: {
+            bool: {
+              should: [
+                ...alertIds.map((id) => {
+                  return {
+                    match_phrase: {
+                      _id: id,
+                    },
+                  };
+                }),
+              ],
```

Comment:
```suggestion
              should: alertIds.map((id) => {
                  return {
                    match_phrase: {
                      _id: id,
                    },
                  };
                }),
```

No need to spread the array :) (suggestion probably has wrong indentation)